src/utilities/mockAuth.js

```diff
@@ -0,0 +1,42 @@
+import store from '../Redux/store';
+
+const signOut = jest.fn();
+const sendPasswordResetEmail = jest.fn((email) => new Promise((resolve, reject) => {
+  if (email === 'bobby@gmail.com') { resolve(true); } reject(Error('Email doesnt exist'));
+}));
+const signInWithEmailAndPassword = jest.fn((email, password) => new Promise((resolve, reject) => {
+  if (password === 'success') { resolve(true); } reject(Error('password is incorrect'));
+}));
+const createUserWithEmailAndPassword = jest.fn((email, password) => new Promise(
+  (resolve, reject) => {
+    if (password === 'success') { resolve(true); } reject(Error('password is invalid'));
+  },
+));
+const updateProfile = jest.fn(({ displayName: name }) => new Promise((resolve, reject) => {
+  if (name === 'success') { resolve(true); } reject(Error('Name is invalid'));
+}));
+const updateEmail = jest.fn((email) => new Promise((resolve, reject) => {
+  if (email === 'success') { resolve(true); } reject(Error('Email is invalid'));
+}));
+const updatePassword = jest.fn((password) => new Promise((resolve, reject) => {
+  if (password === 'success') { resolve(true); } reject(Error('Password is incorrect'));
+}));
+const mockUser = { uid: '1111', displayName: 'Jimbob', email: 'Jimbob@gmail.com' };
+const onAuthStateChanged = jest.fn((success) => (
+  (store.getState().user) ? success(mockUser) : success(null)
+));
+
+const auth = () => ({
```

Comment:
Might be more readable as a giant object instead of having a ton of functions hanging out above (makes this kinda hard to read and debug).

Reply:
I tried to make  giant object, but then they are not being recognized as functions anymore.  Even changing my simplest "signout()" function doesn't work.  I'll show the code and error.
<img width="315" alt="Screen Shot 2020-02-20 at 6 30 51 PM" src="https://user-images.githubusercontent.com/54561452/74995832-30e86280-540f-11ea-8a77-57b9e9b86570.png">
<img width="518" alt="Screen Shot 2020-02-20 at 6 31 03 PM" src="https://user-images.githubusercontent.com/54561452/74995834-33e35300-540f-11ea-9922-c2b73f40964c.png">

Reply:
I bet that your auth methods are getting cleared somehow. That is weird though. I actually don't know how to fix this one (used a similar pattern without problem). Might be worth just changing it back if it's giving you too much of a problem.